### ISSUE TITLE
$ref to the same definition only works on the first property

### ISSUE THREAD

**Issue by tlbdk** [swagger-api/swagger-ui#934]
I'm having an issue with references to the same definition only showing up on the first property, all properties after just get the name of the definition, see the attached screenshot to see the issue.

```js
{
    "swagger": "2.0",
    "info": {
        "version": "1.0.0",
        "title": "Test sample showing swagger error",
        "description": "Show that $ref to the same defination only works on the first property",
        "contact": {
            "name": "Troels Liebe Bentsen",
            "email": "tlb@nversion.dk"
        }
    },
    "host": "localhost",
    "basePath": "/",
    "schemes": [
        "http"
    ],
    "consumes": [
        "application/json"
    ],
    "produces": [
        "application/json"
    ],
    "paths": {
        "/test": {
            "post": {
                "summary": "Test",
                "description": "Test stuff",
                "operationId": "doTest",
                "parameters": [
                    {
                        "in": "body",
                        "name": "body",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/test"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Test stuff response",
                        "schema": {
                          "$ref": "#/definitions/test"
                        }
                    },
                    "405": {
                        "description": "Invalid input"
                    }
                },
                "tags": ["test"]
            }
        }
    },
    "definitions": {
        "test": {
            "required": [
                "test1"
            ],
            "properties": {
                "test1": {
                    "$ref": "#/definitions/badgerfishstring"
                },
                "test2": {
                    "$ref": "#/definitions/badgerfishstring"
                },
                "test3": {
                    "$ref": "#/definitions/badgerfishstring"
                }
            }
        },
        "badgerfishstring": {
            "required": [
                "$"
            ],
            "properties": {
                "$": {
                  "type": "string"
                }
            }
        }
    }
}
```

![image](https://cloud.githubusercontent.com/assets/11635/6191978/00a2fdc6-b3ae-11e4-9dff-2b1407bff246.png)

**Comment by fehguy** [comment]
Are you using swagger-ui 2.1.4-M1?

**Comment by tlbdk** [comment]
Just updated to v2.1.4-M1 and seeing same issue

**Comment by JohanBas** [comment]
I have the same issue with version 1.2

**Comment by ralphvanetten** [comment]
This seems to be a duplicate of #904

**Comment by fehguy** [comment]
this is tested and fixed in 2.1.6-M1 of swagger UI.  tests are here:

https://github.com/swagger-api/swagger-js/blob/c16f45196f4749071c794babae136b45655ec399/test/models.js#L204

Not sure if the 1.2 support can be added.
